src/editor/EditorCtrl.ts

```diff
@@ -157,23 +155,39 @@ export default class EditorCtrl {
               });
             });
           }
-
         }
 
         $scope.togglePaidSection = () => {
+          $scope.draft.hasPaidSection = !$scope.draft.hasPaidSection;
           if ($scope.draft.hasPaidSection) {
-            scrollToLinebreak(undefined, $scope.draft.paidSectionLinebreak);
-          } else {
-            scrollToLinebreak(undefined, 0);
+            if ($scope.draft.paidSectionLinebreak === null) {
+              $scope.draft.paidSectionLinebreak = 1;
+            }
+            checkForCurrencyConversion();
+            const linebreak = $scope.draft.paidSectionLinebreak !== null ? $scope.draft.paidSectionLinebreak : 0;
+            setTimeout(() => {
+              scrollToLinebreak(undefined, linebreak);
+            }, 0);
           }
         }
 
+        const checkForCurrencyConversion = () => {
+          this.walletService.convertBCHtoUSD($scope.draft.paidSectionCost).then((currencies: ICurrencyConversion) => {
+            $scope.$apply(function () {
```

Comment:
We have the scopeService for that. You can firstly set the new param's values and then call the scopeService to apply the changes. In this way you can avoid the callback which is harder to read.